src/filesystem/file_content.rs

```diff
@@ -150,10 +164,15 @@ impl FileContent {
         matches!(self, Self::Stub { .. })
     }
 
+    pub fn is_empty(&self) -> bool {
+        matches!(self, Self::EmptyFile)
+    }
+
     pub fn parse(input: Stream) -> ParserResult<Self> {
         let (input, content_type) = le_u8.parse_peek(input)?;
 
         let parsed = match content_type {
+            FILE_CONTENTY_TYPE_EMPTY => (Stream::default(), FileContent::EmptyFile),
```

Comment:
This definitely doesn't match the encoder. Probably worth adding a round trip test with one of these to make sure the encoder/decoder works.